src/symbol/placement.js

```diff
@@ -83,6 +82,45 @@ export class RetainedQueryData {
     }
 }
 
+class CollisionGroups {
+    collisionGroups: { [layerId: ?string]: { ID: number, predicate?: any }};
+    maxGroupID: number;
+
+    constructor() {
+        this.maxGroupID = 0;
+        this.collisionGroups = {
+            undefined: {
+                ID: 0,
+                predicate: null
+            }
+        };
+    }
+
+    get(groupName?: string) {
+        if (groupName && this.maxGroupID === 0) {
+            // Keep the predicate null until the first collision
+            // group gets added to avoid overhead in the case
+            // everything's in the default group.
+            this.collisionGroups[undefined].predicate =
+                (key) => {
+                    return key.collisionGroup === 0;
+                };
+        }
```

Comment:
This could cause a name collision if a source id (and thus the corresponding `groupName`) were `"undefined"`. Could the default group just be a separate `defaultGroup` property on the class, rather than stored in the `collisionGroups` map?

Reply:
We could then also just reference `defaultGroup` instead of doing `collisionGroups.get()`